ros/src/waypoint_updater/waypoint_updater.py

```diff
@@ -148,29 +191,78 @@ def _closest_waypoint_index(self):
         self.current_waypoint_ahead = prev_index
 
         return prev_index
-        
-    
+
+
     def publish_waypoints_ahead(self):
         """ Publishes a Lane of LOOKAHEAD_WPS waypoints /final_waypoint topic
         """
-        
         if self.base_waypoints is None or self.current_pose is None:
             return
 
         start_index = self._closest_waypoint_index()
         self.current_waypoint_ahead = start_index
 
         waypoint_indices = self._get_waypoint_indices(start_index, LOOKAHEAD_WPS)
-        
+
         lane = Lane()
         lane.header.frame_id = self.current_frame_id
         lane.header.stamp = rospy.Time.now()
         lane.header.seq = self.seq
         lane.waypoints = [self.base_waypoints[i] for i in waypoint_indices]
 
+        if self.closest_obstacle is None or self.current_tfl_state != 2:
+            # There is no traffic light near us, go full speed
+            self.current_velocity = self.current_velocity+0.5 if self.current_velocity < MAX_SPEED else self.current_velocity
+            speeds = np.linspace(self.current_velocity, MAX_SPEED, 1+(LOOKAHEAD_WPS//17))
```

Comment:
hm, I think this calculation will ramp speed up/down based on waypoint indexing, not space between them. for carla and sim the waypoints distances are different

Reply:
Yes, in fact, it does not work well. As the start speed is the current speed but due to delay the car might be at a higher speed, causing deceleration and not acceleration. At the end I fixed the max_speed to let the controller manage the ramp.